### ISSUE TITLE
🐛 BUG: HTML comments are stripped from component output

### ISSUE THREAD

**Issue by thelucid** [snowpackjs/astro#1452]
### What package manager are you using?

yarn

### What operating system are you using?

macOS

### Describe the Bug

HTML comments are stripped from component output. This is problematic as HTML comments are often used as directives for other services. I'm happy to help on addressing this, if there is a good overview of what `.astro` component files are, and how they're rendered. Presumably this would also be the case with React and Vue components.

### Steps to Reproduce

1. `npm init astro` using template blog
2. Add an HTML comment to any component
3. View resulting output
4. HTML comment is gone

**Comment by matthewp** [comment]
This should be fixed in `next`. See [this example](https://astro.build/play#eyJpbm1lbW9yeTovL21vZGVsL3NyYy9wYWdlcy9QYWdlLmFzdHJvIjoiLS0tXG5pbXBvcnQge2Zvcm1hdH0gZnJvbSAnZGF0ZS1mbnMnOyBcblxuLy8gV2VsY29tZSB0byBBc3RybyFcbi8vIFdyaXRlIEphdmFTY3JpcHQgJiBUeXBlU2NyaXB0IGhlcmUsIGluIHRoZSBcImNvbXBvbmVudCBzY3JpcHQuXCJcbi8vIFRoaXMgd2lsbCBydW4gZHVyaW5nIHRoZSBidWlsZCwgYnV0IG5ldmVyIGluIHRoZSBmaW5hbCBvdXRwdXQuXG4vLyBVc2UgdGhlc2UgdmFyaWFibGVzIGluIHRoZSBIVE1MIHRlbXBsYXRlIGJlbG93LlxuLy9cbi8vIEZ1bGwgU3ludGF4OlxuLy8gaHR0cHM6Ly9kb2NzLmFzdHJvLmJ1aWxkL2NvcmUtY29uY2VwdHMvYXN0cm8tY29tcG9uZW50cy9cblxuY29uc3QgYnVpbHRBdDogRGF0ZSA9IG5ldyBEYXRlKCk7XG5jb25zdCBidWlsdEF0Rm9ybWF0dGVkID0gZm9ybWF0KGJ1aWx0QXQsICdNTU1NIGRkLCB5eXl5IC0tIEg6bW06c3MuU1NTJyk7XG4tLS1cbjxodG1sIGxhbmc9XCJlblwiPlxuICA8aGVhZD5cblxuICA8L2hlYWQ+XG4gIDxib2R5PlxuICAgIDxoZWFkZXI+XG4gICAgICA8IS0tIENvbW1lbnQgaGVyZS0tPlxuICAgICAgPGgxPkNvbW1lbnRzIGluY2x1ZGVkPC9oMT5cbiAgICA8L2hlYWRlcj5cbiAgPC9ib2R5PlxuPC9odG1sPlxuIn0=).

Going to leave open until that branch is merged.

**Comment by thelucid** [comment]
That's great news, thanks @matthewp.

**Comment by natemoo-re** [comment]
Following up on Matthew's comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
